src/cargo/lib.rs

```diff
@@ -142,7 +142,8 @@ pub fn exit_with_error(err: CliError, shell: &mut Shell) -> ! {
         } else if fatal {
             drop(shell.error(&error))
         } else {
-            drop(writeln!(shell.err(), "{}", error))
+            // Non-fatal messages, like --help output, go to stdout.
```

Comment:
I believe this comment is a bit misleading: a ton of non-fatal messages, like `Downloading foo...` purposefully use stderr. Maybe drop this comment or change it to "unlike all other human-readable output, help goes to stdout so as to make it easier to use pager with cargo --help".

Reply:
Makes sense.  I'll just drop it, I suppose anybody who is curious can read the history.